File: 2448-count-number-of-bad-pairs/count-number-of-bad-pairs.java
class Solution {
    public long countBadPairs(int[] nums) {
        long badPairs = 0;
        int n = nums.length;
        
        // Calculate total pairs (i, j) where i < j
        long totalPairs = (long) n * (n - 1) / 2;
        
        // Map to store the count of (nums[j] - j)
        Map<Integer, Integer> map = new HashMap<>();
        
        for (int j = 0; j < n; j++) {
            int diff = nums[j] - j;
            if (map.containsKey(diff)) {
                badPairs += (j - map.get(diff));
            } else {
                badPairs += j;
            }
            map.put(diff, map.getOrDefault(diff, 0) + 1);
        }
        
        // Return the total number of bad pairs
        return badPairs;
    }
}

File: 2448-count-number-of-bad-pairs/count-number-of-bad-pairs.py
class Solution(object):
    def countBadPairs(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        n = len(nums)
        bad_pairs = 0
        
        # Dictionary to store the frequency of nums[j] - j
        freq = {}
        
        for j in range(n):
            diff = nums[j] - j
            
            # Calculate the number of good pairs
            if diff in freq:
                bad_pairs += j - freq[diff]
                freq[diff] += 1
            else:
                bad_pairs += j
                freq[diff] = 1
        
        return bad_pairs
